src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs

```diff
@@ -156,122 +152,49 @@ internal partial class SQLitePersistentStorage : AbstractPersistentStorage
                 _shutdownTokenSource.Token);
         }
 
-        private SqlConnection GetConnection()
+        public static SQLitePersistentStorage? TryCreate(
+            SQLiteConnectionPoolService connectionPoolService,
+            Solution? bulkLoadSnapshot,
+            string workingFolderPath,
+            string solutionFilePath,
+            string databaseFile,
+            IPersistentStorageFaultInjector? faultInjector)
         {
-            lock (_connectionGate)
+            var sqlStorage = new SQLitePersistentStorage(connectionPoolService, bulkLoadSnapshot, workingFolderPath, solutionFilePath, databaseFile, faultInjector);
+            if (sqlStorage._connectionPool is null)
             {
-                // If we have an available connection, just return that.
-                if (_connectionsPool.Count > 0)
-                {
-                    return _connectionsPool.Pop();
-                }
+                // The connection pool failed to initialize
+                return null;
             }
 
-            // Otherwise create a new connection.
-            return SqlConnection.Create(_faultInjector, DatabaseFile);
+            return sqlStorage;
         }
 
-        private void ReleaseConnection(SqlConnection connection)
+        public override void Dispose()
         {
-            lock (_connectionGate)
+            try
             {
-                // If we've been asked to shutdown, then don't actually add the connection back to 
-                // the pool.  Instead, just close it as we no longer need it.
-                if (_shutdownTokenSource.IsCancellationRequested)
-                {
-                    connection.Close_OnlyForUseBySqlPersistentStorage();
-                    return;
-                }
-
+                // Flush all pending writes so that all data our features wanted written are definitely
+                // persisted to the DB.
                 try
                 {
-                    _connectionsPool.Push(connection);
+                    FlushWritesOnClose();
                 }
-                catch
+                catch (Exception e)
                 {
-                    // An exception (likely OutOfMemoryException) occurred while returning the connection to the pool.
-                    // The connection will be discarded, so make sure to close it so the finalizer doesn't crash the
-                    // process later.
-                    connection.Close_OnlyForUseBySqlPersistentStorage();
-                    throw;
+                    // Flushing may fail.  We still have to close all our connections.
```

Comment:
consider mentioning who will do this.   note: i haven't 100% convinced myself this works/is-desirable.

Reply:
Not making changes to that since I wasn't touching this code (aside from a formatting-required indentation adjustment).